pgxn/remotexact/remotexact.c

```diff
@@ -95,6 +117,9 @@ init_rwset_collection_buffer(Oid dbid)
 		return;
 	}
 
+	// Before collecting the rwset, mark the xact as remote in the procarray.
+	set_remotexact_procflags();
```

Comment:
The `init_rwset_collection_buffer` is called for every transaction so this is going to mark all transactions as remote.

A better place to do this is inside `rwset_add_region`, where you can check if more than 2 regions are added.

Reply:
Done.

Reply:
I have added the change in the validate_and_apply_xact as well. 

I didn't reuse the function because we need the code to be inline.